test/libyul/objectCompiler/eof/prune_unreferenced_container.yul

```diff
@@ -0,0 +1,81 @@
+object "a" {
+    code {
+        mstore(0, eofcreate("c", 0, 0, 0, 0))
+        return(0, 32)
+    }
+
+    object "b" {
+        code {
+            mstore(0, 0x1122334455667788990011223344556677889900112233445566778899001122)
+            mstore(32, 0x1122334455667788990011223344556677889900112233445566778899001122)
+        }
+    }
+
+    object "c" {
+        code {
+            mstore(0, 0x1122334455667788990011223344556677889900112233445566778899001122)
+            mstore(32, 0x1122334455667788990011223344556677889900112233445566778899001122)
+        }
+    }
+}
+
+// ====
+// EVMVersion: >=prague
+// bytecodeFormat: >=EOFv1
+// ----
+// Assembly:
+//     /* "source":80:81   */
+//   0x00
+//     /* "source":56:82   */
+//   dup1
+//   dup1
+//   dup1
+//   eofcreate{1}
+//     /* "source":53:54   */
+//   0x00
+//     /* "source":46:83   */
+//   mstore
+//     /* "source":106:108   */
+//   0x20
+//     /* "source":103:104   */
+//   0x00
+//     /* "source":96:109   */
+//   return
+// stop
+//
+// sub_0: assembly {
```

Comment:
Maybe we should be removing unreferenced containers somewhere before assembling after all. It's actually a bit weird that we show them in assembly but then they're missing from the bytecode. It makes the assembly less reliable as way for users to inspect what is being generated (and for me to verify that the expectations are actually correct because the change is only in the binary :)).

Of course it's not something to address in this PR, but maybe something to think about. The problem is actually similar to removing unreferenced functions from CFG in a way.

Reply:
I would also remove the subcontainers earlier. On CFG level ideally.